tools/ncf_constraints.py

```diff
@@ -0,0 +1,90 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+
+### Constraint checking function 
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return parameter.length <= max_size
+
+def min_length( parameter, min_size):
+  return parameter.length >= min_size
+
+def match_regexp(parameter, regexp):
+  match = re.match(regexp, parameter, re.S)
```

Comment:
Why have you used the ```re.S``` option here?

I know that is causes the regexp expression ```.``` to match newlines like any other character. What brought you to choose to do this here? I think we should at least write a comment to explain, and certainly document it somewhere.